statesync/reactor.go

```diff
@@ -25,202 +31,263 @@ const (
 
 // Reactor handles state sync, both restoring snapshots for the local node and serving snapshots
 // for other nodes.
+//
+// TODO: Remove any fields or embedded types that are no longer needed once the
+// new p2p router is fully implemented.
+// ref: https://github.com/tendermint/tendermint/issues/5670
 type Reactor struct {
-	p2p.BaseReactor
+	// XXX: Keep a reference to the legacy p2p Switch in order to implement
+	// MessageValidator and maintain current p2p Receive behavior.
+	p2pSwitch *p2p.Switch
 
-	conn      proxy.AppConnSnapshot
-	connQuery proxy.AppConnQuery
-	tempDir   string
+	logger      log.Logger
+	conn        proxy.AppConnSnapshot
+	connQuery   proxy.AppConnQuery
+	tempDir     string
+	snapshotCh  *p2p.Channel
+	chunkCh     *p2p.Channel
+	peerUpdates p2p.PeerUpdates
 
-	// This will only be set when a state sync is in progress. It is used to feed received
-	// snapshots and chunks into the sync.
+	// This will only be set when a state sync is in progress. It is used to feed
+	// received snapshots and chunks into the sync.
 	mtx    tmsync.RWMutex
 	syncer *syncer
 }
 
-// NewReactor creates a new state sync reactor.
-func NewReactor(conn proxy.AppConnSnapshot, connQuery proxy.AppConnQuery, tempDir string) *Reactor {
-	r := &Reactor{
-		conn:      conn,
-		connQuery: connQuery,
+// NewReactor returns a reference to a new state-sync reactor. It accepts a Context
+// which will be used for cancellations and deadlines when executing the reactor,
+// a p2p Channel used to receive and send messages, and a router that will be
+// used to get updates on peers.
+func NewReactor(
+	logger log.Logger,
+	p2pSwitch *p2p.Switch,
+	conn proxy.AppConnSnapshot,
+	connQuery proxy.AppConnQuery,
+	snapshotCh, chunkCh *p2p.Channel,
+	peerUpdates p2p.PeerUpdates,
+	tempDir string,
+) *Reactor {
+	return &Reactor{
+		logger:      logger,
+		p2pSwitch:   p2pSwitch,
+		conn:        conn,
+		connQuery:   connQuery,
+		snapshotCh:  snapshotCh,
+		chunkCh:     chunkCh,
+		peerUpdates: peerUpdates,
+		tempDir:     tempDir,
 	}
-	r.BaseReactor = *p2p.NewBaseReactor("StateSync", r)
-	return r
 }
 
-// GetChannels implements p2p.Reactor.
-func (r *Reactor) GetChannels() []*p2p.ChannelDescriptor {
-	return []*p2p.ChannelDescriptor{
-		{
-			ID:                  SnapshotChannel,
-			Priority:            3,
-			SendQueueCapacity:   10,
-			RecvMessageCapacity: snapshotMsgSize,
-		},
-		{
-			ID:                  ChunkChannel,
-			Priority:            1,
-			SendQueueCapacity:   4,
-			RecvMessageCapacity: chunkMsgSize,
-		},
-	}
+func (r *Reactor) setConn(conn proxy.AppConnSnapshot) {
```

Comment:
This was added to address the concurrency patterns of the testing suite and the reactor's use of `conn`. Since I'm reverting the use of suite, this won't be necessary anymore.